### PR TITLE
Document empty or nil prometheus service/pod monitors

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: apis/v1alpha1/opentelemetrycollector_types.go [hunk 1]
```diff
@@ -380,11 +380,13 @@ type OpenTelemetryTargetAllocatorPrometheusCR struct {
 	// PodMonitors to be selected for target discovery.
 	// This is a map of {key,value} pairs. Each {key,value} in the map is going to exactly match a label in a
 	// PodMonitor's meta labels. The requirements are ANDed.
+	// Empty or nil map matches all pod monitors.
```
Comment (jaronoff97) [inline apis/v1alpha1/opentelemetrycollector_types.go:383]: an explicit nil means select nothing iirc.

**Review by jaronoff97 — COMMENTED** [review]
i think we need a make generate here for the docs/types

File: apis/v1alpha1/opentelemetrycollector_types.go (hunk 1 above)
Comment (swiatekm-sumo) [reply apis/v1alpha1/opentelemetrycollector_types.go:383]: It doesn't, there's no way to set nothing with these label maps. I think you're mixing this up with proper LabelSelectors in `v1alpha2`.

**Comment by pavolloffay** [comment]
> i think we need a make generate here for the docs/types

`make api-docs bundle  generate ` does not make any change

**Comment by pavolloffay** [comment]
And all checks are passing

File: apis/v1alpha1/opentelemetrycollector_types.go (hunk 1 above)
Comment (pavolloffay) [reply apis/v1alpha1/opentelemetrycollector_types.go:383]: @swiatekm-sumo so is it correct? Can it be merged or do I need to make any change?

Comment (jaronoff97) [reply apis/v1alpha1/opentelemetrycollector_types.go:383]: 🤦 i am mixing those up.

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**